### ISSUE TITLE
could not broadcast input array from shape (4,260,260) into shape (4)

### ISSUE THREAD

**Issue by LXYTSOS** [milesial/Pytorch-UNet#26]
when I run the train.py I got this error at `imgs = np.array([i[0] for i in b]).astype(np.float32)`(train.py line 65), by the way, my training images are not share the same width and height size.

**Comment by milesial** [comment]
Hi, can you check if the images are sucessfully preprocessed ? (croped, normalized, ...)
The function that modifies the generator is https://github.com/milesial/Pytorch-UNet/blob/1e8093de0398364cc65f3c7722b61a0f560df104/utils/load.py#L29

The width and height should be fine as long as the width is greater than the height !

**Comment by LXYTSOS** [comment]
I've already solved this problem, it's because my training images are png files that have 4 channels, I converted them into 3 channels and solved it.

**Comment by karlita101** [comment]
> m, it's because my training images are png files that have 4 channels, I converted them into 3 channels and solved it.

Hi there, were you not able to get it to work with 4 channels? I am currently working with png files too and would like to use the alpha channel information but can't seem to get it to work



`raceback (most recent call last):
  File "train.py", line 189, in <module>
    val_percent=args.val / 100)
  File "train.py", line 113, in train_net
    writer.add_images('images', imgs, global_step)
  File "/home/karla2021/anaconda3/envs/unetpytorch/lib/python3.7/site-packages/torch/utils/tensorboard/writer.py", line 589, in add_images
    image(tag, img_tensor, dataformats=dataformats), global_step, walltime)
  File "/home/karla2021/anaconda3/envs/unetpytorch/lib/python3.7/site-packages/torch/utils/tensorboard/summary.py", line 376, in image
    tensor = convert_to_HWC(tensor, dataformats)
  File "/home/karla2021/anaconda3/envs/unetpytorch/lib/python3.7/site-packages/torch/utils/tensorboard/_utils.py", line 107, in convert_to_HWC
    tensor_CHW = make_grid(tensor_NCHW)
  File "/home/karla2021/anaconda3/envs/unetpytorch/lib/python3.7/site-packages/torch/utils/tensorboard/_utils.py", line 76, in make_grid
    assert I.ndim == 4 and I.shape[1] == 3
`
